### ISSUE TITLE
A model for overflowing content on scroll

### ISSUE THREAD

**Issue by alanorozco** [ampproject/amphtml#21044]
## Background

Re: #20881 on exposing `scrollUp/down` events:

> I have a use case for https://github.com/ampproject/amphtml/issues/20125, which has been closed in favor of this issue. For context, it's what I outlined here: https://github.com/ampproject/amphtml/issues/21016#issuecomment-466546960
> 
> ![img_0276](https://user-images.githubusercontent.com/1528181/53273935-56cb1200-36aa-11e9-83eb-fab123a1bfe5.jpg)
> 
> [I can partially implement this with position observer, amp-animation, `translateY` and a position](https://codepen.io/cathyxz/pen/YBmKBq). So if we remove the animation exit, and instead hide this sticky footer on scrollDown, I think that should get us all of the way there. 
> 
> Is `scrollUp` and `scrollDown` still something we're looking at adding? Could they be used to trigger animations? 

Most of the drawbacks with exposing the events still exist:

- `amp-animation` is large (vast majority of it is unneeded) and awkward for this use case.
- Overloading is bad. Letting users attach whatever to scroll events is hard to limit.
- If this is a common use-case this is much better wrapped in a model that's easy to use.

Specific to this use-case is the drawback that this must call `changeHeight` in which case valid uses would be hard to detect.

## Goal

Provide a set of APIs encapsulating a **model for overflowing content on scroll**.

## Details

This makes sense to me as an addition to `amp-position-observer`.

Some things we can enforce:

* That overflowing content doesn't have any more than `FACTOR * 100vh` under it which should probably be something like half a viewport.

* That this triggers on exactly the right scroll offset.

* API niceness from hooking into overflow element for toggling.

/cc @ampproject/wg-ui-and-a11y @cathyxz @aghassemi

**Comment by aghassemi** [comment]
Closing in favor of alt solution which as been implemented (https://github.com/ampproject/amphtml/issues/20881). 

Please reopen if I am wrong here.
